test/CheckedC/dump-dataflow-facts.c

```diff
@@ -0,0 +1,256 @@
+// Tests for dumping of datafow analysis for collecting facts
+//
+// RUN: %clang_cc1 -fcheckedc-extension -fdump_extracted_comparison_facts %s 2>1 | FileCheck %s
+
+//===================================================================
+// Dumps of different kinds of collected dataflow facts
+//===================================================================
+
+
+int f(int a);
+
+void fn_1(void) {
+  int a, b, c;
+  if (b < c)
+    if (f((a=5)+3))
+      b = c;
+}
+
+// CHECK: Block #4: {
+// CHECK: In:
```

Comment:
Can you use CHECK-NEXT instead of CHECK here? Also inside a new function it's good to check for the function name, like:

`CHECK-LABEL: fn_1`

Reply:
@mgrang Why should I check for the function name?

Reply:
> @mgrang Why should I check for the function name?

To make sure you are checking output from the correct function. Consider this:

```
// CHECK: Block #1 
// CHECK: In

void foo() {}
void bar() {}
```

Here it is not clear whose output you are checking. The output of foo may conflict with the output of bar and can confuse lit. So it's good to always check for `CHECK-LABEL` first. See [https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-label-directive](url)

Reply:
@mgrang Do I need to dump the function name as well to be able to use CHECK-LABEL?
I followed the doc on this directive but it cannot match the function name inside the function.

Reply:
I see you are not dumping the IR. Then you do not need CHECK-LABEL, I guess. However, would it be more useful to the user to also dump the function name as part of your new flag?

Reply:
I think it makes the future debugging easier if the function name is dumped.

Reply:
I think so too. So once you add it to your flag you can add back the CHECK-LABELs.

Reply:
Good idea, thanks!